pex/expose_scripts.py

```diff
@@ -0,0 +1,70 @@
+import hashlib
+import itertools
+import os
+
+from pathlib import Path
+
+
+SCRIPT_TEMPLATE = """#!/bin/sh
+
+{pex_var}={entry} exec {executable} {pex_path} $@
+"""
+
+
+def get_hash(shebang, pex_path, entry_points, scripts):
+    entry_points = sorted(entry_points)
+    scripts = sorted(scripts)
+
+    hasher = hashlib.md5()
+    hasher.update(shebang.encode("utf-8"))
+    hasher.update(pex_path.encode("utf-8"))
+    for item in itertools.chain(entry_points, scripts):
```

Comment:
This hashing scheme is foiled by setting an entry_point for the module 'pex' today and then changing that to be the 'pex' console script tomorrow. You might form a dict using the key names corresponding to the 4 slots and then hash its sorted key json encoding.